src/functions-and-arrays.js

```diff
@@ -26,13 +72,33 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+  let roundedSum = 0;
+  if (numbersAvg.length === 0) {
+    return null;
+  } else {
+    for (let k = 0; k<numbersAvg.length; k++) {
+      if (numbersAvg.length === 1) {
+        roundedSum = (roundedSum + numbersAvg[k]) / numbersAvg.length;
+      }
+      else {
+        roundedSum = (roundedSum + numbersAvg[k]) / numbersAvg.length;
+      }
+    }
+    return roundedSum;
+  }
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+  let averageWordsArr = "";
+  if (wordsArr.length === 0) {
+    return null;
```

Comment:
In this iteration, the logic is close to the previous one. The main difference is that when we loop over the array, we want to get the `length` of the word that we are iterating over.
Something like this:
```
function averageWordLength(wordsArr) {
  if (wordsArr.length === 0) {
    return null;
    }
  let total = 0;
  for (let i = 0; i < wordsArr.length; i++) {
    total += wordsArr[i].length;
  }
  let average = total / wordsArr.length;
  return average;
}
```